packer/scripts/Artifacts/MsSql/Structure/Ods/2050-CreateTableValueFunctions.sql

```diff
@@ -0,0 +1,574 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE OR ALTER FUNCTION reporting.GetFullTimeEquivalency
+(
+    @searchDate DATE
+)
+RETURNS @ResultTable TABLE
+(
+    LocalEducationAgencyId INT,
+    StudentUSI INT,
+    Total_FTE INT,
+    INDEX idx_fte CLUSTERED(LocalEducationAgencyId, StudentUSI)
+)
+AS
+BEGIN
+    INSERT INTO @ResultTable (LocalEducationAgencyId, StudentUSI, Total_FTE)
+    SELECT
+        s.LocalEducationAgencyId,
+        ssa.StudentUSI,
+        total_fte = SUM(ssa.FullTimeEquivalency)
+    FROM edfi.StudentSchoolAssociation ssa
+    INNER JOIN sk.StudentSchoolAssociationExtension ssae ON
+        ssa.StudentUsi = ssae.StudentUSI
+        AND ssa.EntryDate = ssae.EntryDate
+        AND ssa.SchoolId = ssae.SchoolId
+    INNER JOIN edfi.school s ON
+        ssae.ReportingSchoolId = s.SchoolId
+    WHERE
+        (ssa.ExitWithdrawDate IS NULL OR ssa.ExitWithdrawDate >= @searchDate)
+        AND ssa.EntryDate <= @searchDate
+    GROUP BY
+        s.LocalEducationAgencyId,
+        ssa.StudentUSI;
+
+    RETURN;
+END;
+GO
+
+CREATE OR ALTER FUNCTION reporting.GetMaxFullTimeEquivalency
+(
+    @searchDate DATE
+)
+RETURNS @ResultTable TABLE
+(
+    LocalEducationAgencyId INT,
+    StudentUSI INT,
+    Total_FTE INT,
+    Max_FTE INT,
+    INDEX idx_maxfte CLUSTERED(LocalEducationAgencyId, StudentUSI)
+)
+AS
+BEGIN
+    INSERT INTO @ResultTable (LocalEducationAgencyId, StudentUSI, Total_FTE, Max_FTE)
+    SELECT
+        ssa.StudentUSI,
+        med.LocalEducationAgencyId,
+        med.Total_FTE,
+        Max_FTE = max(ssa.FullTimeEquivalency)
+    FROM edfi.StudentSchoolAssociation ssa
+    INNER JOIN sk.StudentSchoolAssociationExtension ssae ON
+        ssa.StudentUSI = ssae.StudentUSI
+        AND ssa.SchoolId = ssae.SchoolId
+        AND ssa.EntryDate = ssae.EntryDate
+    INNER JOIN edfi.School sch_Reporting ON
+        ssae.ReportingSchoolId = sch_Reporting.SchoolId
+    INNER JOIN reporting.GetFullTimeEquivalency(@searchDate) med ON
+        ssa.StudentUSI = med.StudentUSI
+        AND sch_Reporting.LocalEducationAgencyId = med.LocalEducationAgencyId
+    INNER JOIN edfi.Descriptor d_gradelevel ON
+        ssa.EntryGradeLevelDescriptorId = d_gradelevel.DescriptorId
+    WHERE
+        med.total_fte >= CASE WHEN d_gradelevel.CodeValue IN ('hp', 'pk') THEN 0 else 51 END
```

Comment:
@jamessmckay codes 'hp' and 'pk' do not exist in the code descriptors. Need to figure out what they represent.